projects/igniteui-angular/src/lib/services/exporter-common/base-export-service.ts

```diff
@@ -1,12 +1,34 @@
 import { EventEmitter } from '@angular/core';
 
-import { cloneValue, IBaseEventArgs, resolveNestedPath, yieldingLoop } from '../../core/utils';
+import { cloneArray, cloneValue, IBaseEventArgs, resolveNestedPath, yieldingLoop } from '../../core/utils';
 import { DataUtil } from '../../data-operations/data-util';
 
 import { ExportUtilities } from './export-utilities';
 import { IgxExporterOptionsBase } from './exporter-options-base';
 import { ITreeGridRecord } from '../../grids/tree-grid/tree-grid.interfaces';
 import { TreeGridFilteringStrategy } from '../../grids/tree-grid/tree-grid.filtering.pipe';
+import { IGroupingState } from '../../data-operations/groupby-state.interface';
+import { getHierarchy, isHierarchyMatch } from '../../data-operations/operations';
+import { IGroupByExpandState } from '../../data-operations/groupby-expand-state.interface';
+import { IFilteringState } from '../../data-operations/filtering-state.interface';
+import { IgxGridBaseDirective } from '../../grids/public_api';
+import { IgxTreeGridComponent } from '../../grids/tree-grid/public_api';
+import { IgxGridComponent } from '../../grids/grid/public_api';
+import { DatePipe } from '@angular/common';
+import { IGroupByRecord } from '../../data-operations/groupby-record.interface';
+
+export enum RecordType {
+    GroupedRecord = 1,
+    TreeGridRecord = 2,
+    DataRecord = 3,
+}
+
+export interface IExportRecord {
+    data: any;
+    level: number;
+    hidden?: boolean;
+    type: number;
```

Comment:
Why not `RecordType`?
```suggestion
    type: RecordType;
```